src/BenchmarkDotNet.Core/Portability/Cpu/SysctlCpuInfoParser.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections.Generic;
+using BenchmarkDotNet.Extensions;
+using BenchmarkDotNet.Helpers;
+using JetBrains.Annotations;
+
+namespace BenchmarkDotNet.Portability.Cpu
+{
+    public class SysctlCpuInfoParser : ICpuInfo
```

Comment:
For `ProcCpuInfoParser`, `WmicCpuInfoParser`, `SysctlCpuInfoParser`, it's better to split parsing logic and results.
I suggest writing something like this:
```cs
internal static class SysctlCpuInfoParser
{
    [NotNull]
    internal static ICpuInfo Parse([CanBeNull] string content)
    {
        var sysctl = SectionsHelper.ParseSection(content, ':');
        var processorName = sysctl.GetValueOrDefault("machdep.cpu.brand_string");
        var physicalProcessorCount = GetPositiveIntValue(sysctl, "hw.packages");
        var physicalCoreCount = GetPositiveIntValue(sysctl, "hw.logicalcpu");
        var logicalCoreCount = GetPositiveIntValue(sysctl, "hw.physicalcpu");
        return new CpuInfo(processorName, physicalProcessorCount, physicalCoreCount, logicalCoreCount);
    }


    [CanBeNull]
    private static int? GetPositiveIntValue([NotNull] Dictionary<string, string> sysctl, string keyName)
    {
        if (sysctl.TryGetValue(keyName, out string value) &&
            int.TryParse(value, out int result) &&
            result > 0)
            return result;
        return null;
    }
}

public class CpuInfo : ICpuInfo
{
    public string ProcessorName { get; }
    public int? PhysicalProcessorCount { get; }
    public int? PhysicalCoreCount { get; }
    public int? LogicalCoreCount { get; }

    public CpuInfo(string processorName, int? physicalProcessorCount, int? physicalCoreCount, int? logicalCoreCount)
    {
        ProcessorName = processorName;
        PhysicalProcessorCount = physicalProcessorCount;
        PhysicalCoreCount = physicalCoreCount;
        LogicalCoreCount = logicalCoreCount;
    }
}
```

Reply:
@AndreyAkinshin ok, thanks